plugins/aggregators/valuecounter/valuecounter.go

```diff
@@ -0,0 +1,109 @@
+package valuecounter
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/aggregators"
+)
+
+type aggregate struct {
+	name       string
+	tags       map[string]string
+	fieldCount map[string]int
+}
+
+// ValueCounter an aggregation plugin
+type ValueCounter struct {
+	cache  map[uint64]aggregate
+	Fields []string
+}
+
+// NewValueCounter create a new aggregation plugin which counts the occurances
+// of fields and emits the count.
+func NewValueCounter() telegraf.Aggregator {
+	vc := &ValueCounter{}
+	vc.Reset()
+	return vc
+}
+
+var sampleConfig = `
+  ## General Aggregator Arguments:
+  ## The period on which to flush & clear the aggregator.
+  period = "30s"
+  ## If true, the original metric will be dropped by the
+  ## aggregator and will not get sent to the output plugins.
+  drop_original = false
+  ## The fields for which the values will be counted
+  fields = []
+`
+
+// SampleConfig generates a sample config for the ValueCounter plugin
+func (vc *ValueCounter) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns the description of the ValueCounter plugin
+func (vc *ValueCounter) Description() string {
+	return "Count the occurance of values in fields."
+}
+
+// Add is run on every metric which passes the plugin
+func (vc *ValueCounter) Add(in telegraf.Metric) {
+	id := in.HashID()
+
+	// Check if the cache already has an entry for this metric, if not create it
+	if _, ok := vc.cache[id]; !ok {
+		a := aggregate{
+			name:       in.Name(),
+			tags:       in.Tags(),
+			fieldCount: make(map[string]int),
+		}
+		vc.cache[id] = a
+	}
+
+	// Check if this metric has fields which we need to count, if so increment
+	// the count.
+	for fk, fv := range in.Fields() {
+		for _, cf := range vc.Fields {
+			if fk == cf {
+				// Do not process float types to prevent memory from blowing up
+				switch fv.(type) {
+				default:
+					log.Printf("I! Valuecounter: Unsupported field type. " +
+						"Must be an int, string or bool. Ignoring.")
+					continue
+				case uint8, uint16, uint32, uint64, int8,
+					int16, int32, int64, string, bool:
```

Comment:
Only need to support int64, uint64, bool, and string.

Reply:
ok, thanks, updated.